src/Resources/public/js/panel.js

```diff
@@ -257,15 +258,14 @@ pimcore.plugin.objectmerger.panel = Class.create({
             var leftValue;
 
             var changed = false;
-
             if (item) {
-                leftValue = item.value;
+                leftValue = (item.type && item.type == "date")?this.getFormattedDateFromTimestamp(item.data):item.value;
 
                 if (this.resultData[item.key]) {
                     rightValue = this.resultData[item.key].value;
                     changed = this.resultData[item.key].changed;
                 } else {
-                    rightValue = item.value2;
+                    rightValue = (item.type && item.type == "date")?this.getFormattedDateFromTimestamp(item.data):item.value2;
```

Comment:
```suggestion
                    rightValue = (item.type && item.type == "date")?this.getFormattedDateFromTimestamp(item.data2):item.value2;
```
also it's covering only `date`, but doesn't work for `datetime`